test/tests/functional/pbs_mom_hook_sync.py

```diff
@@ -0,0 +1,353 @@
+
+        rc = self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostB)
+        self.assertEqual(rc, 0)
+
+        self.hook_name = "cpufreq"
+        hook_body = "import pbs\n"
+        a = {'event': 'execjob_begin', 'enabled': 'True'}
+        self.server.create_import_hook(self.hook_name, a, hook_body)
+
+        hook_config = """{
+    "apple"         : "pears",
+    "banana"         : "cucumbers"
+}
+"""
+        (fd, fn) = self.du.mkstemp()
+        os.write(fd, hook_config)
+        os.close(fd)
+        a = {'content-type': 'application/x-config',
+             'content-encoding': 'default',
+             'input-file': fn}
+        self.server.manager(MGR_CMD_IMPORT, HOOK, a, self.hook_name)
+        os.remove(fn)
+        self.momA.log_match('cpufreq.HK;copy hook-related ' +
+                            'file request received',
+                            max_attempts=10)
+
+        self.momA.log_match('cpufreq.PY;copy hook-related ' +
+                            'file request received',
+                            max_attempts=10)
+
+        self.momA.log_match('cpufreq.CF;copy hook-related ' +
+                            'file request received',
+                            max_attempts=10)
+
+        self.momB.log_match('cpufreq.HK;copy hook-related ' +
+                            'file request received',
+                            max_attempts=10)
+
+        self.momB.log_match('cpufreq.PY;copy hook-related ' +
+                            'file request received',
+                            max_attempts=10)
+
+        self.momB.log_match('cpufreq.CF;copy hook-related ' +
+                            'file request received',
+                            max_attempts=10)
+
+    def tearDown(self):
+        self.momB.signal("-CONT")
+        self.momB.stop()
+        TestFunctional.tearDown(self)
+
+    def test_1(self):
+        """
+        Given an existing mom hook, suspend mom on hostB,
+        change the hook to be a server hook (causes a
+        delete action), then change it back to a mom hook
+        (results in a send action), and then resume mom.
+        The delete action occurs first and then the send
+        action so we end up with a mom hook in place.
+        """
+
+        self.momB.signal('-STOP')
+
+        # Turn current mom hook into a server hook
+        self.server.manager(MGR_CMD_SET, HOOK,
+                            {'event': 'queuejob'},
+                            id=self.hook_name)
+
+        # Turn current mom hook back to a mom hook
+        self.server.manager(MGR_CMD_SET, HOOK,
+                            {'event': 'exechost_periodic'},
+                            id=self.hook_name)
+
+        # For testability, delay resuming the mom so we can
+        # get a different timestamp on the hook updates
+        time.sleep(3)
+
+        now = int(time.time())
+        self.momB.signal('-CONT')
+
+        # Delay again so we can get all the matches
+        time.sleep(3)
```

Comment:
Please log a message before you sleep indicating the reason. There are other occurrences of this.

Reply:
It would be better to actually log a message like...
self.logger.info("Waiting 3 seconds for operation to complete")
That would help to explain the delay for anyone watching the output or reviewing timestamps in the log.

Reply:
Ok, I'll add the logging call.